src/LogicTasks/Semantics/Decide.hs

```diff
@@ -153,12 +179,19 @@ completeGrade DecideInst{..} sol = do
       translate $ do
         german $ "Die Lösung beinhaltet " ++ display ++ " Fehler."
         english $ "Your solution contains " ++ display ++ " mistakes."
```

Comment:
Similar consideration here. I'm pretty sure that the code originally relied on the fact that `diff /= 0` is only "commented on in the feedback" when we already know that `solLen == acLen`, so that "mistakes" really means "the wrong indices chosen, but neither too many nor too few". This is not the case anymore now, after `preventWithHint -> printHint` in the preceding two blocks.

Reply:
`diff` actually says how many of the provided unique indices are wrong. I had to do some refactoring after I found out, that the two `printWithHint` only check for unique indices. So, for example, `[1,1,1,1,1,1]` did not have enough indices even though there are six wrong rows. I changed the feedback in order to reflect that. Here are some sample outputs:
![image](https://github.com/fmidue/logic-tasks/assets/24428341/380c56fb-949e-49a4-abec-aa1e0711a506)
![image](https://github.com/fmidue/logic-tasks/assets/24428341/78355fbf-21a9-4dc8-8408-5f88c065b678)